invenio_records_resources/services/records/service.py

```diff
@@ -415,3 +415,39 @@ def rebuild_index(self, identity, uow=None):
                 self.indexer.index(rec)
 
         return True
+
+    #
+    # notification handlers
+    #
+    def on_relation_update(self, identity, record_type, records, notif_time):
+        """Handles the update of a related field record."""
+        relations = self.record_cls.relations
+        # fieldpaths = relations.inverse_map.get(record_type)
+        fieldpaths = ["metadata.inner_record"]
+
+        clauses = []
+        for field in fieldpaths:
+            for record in records:
+                recid = record["id"]
+                revision_id = record["revision_id"]
+                clauses.append(Q("bool", must=[
+                    Q("term", **{f"{field}.id": recid}),
+                    # Q("term", **{f"{field}.@v": f"!='{recid}:{revision_id}'"}) # noqa
+                ]))
+
+        filter = [Q("range", indexed_at={"lte": notif_time})]
+        es_query = Q(
+            "bool", minimum_should_match=1, should=clauses, filter=filter
+        )
+
+        search = self.create_search(
```

Comment:
Assuming the reindex method doesn't work properly? if so we can simply also change it to what you need it to do. I don't think it's really being used anywhere

Reply:
I will look into it, it was mainly to be able to build the query with ES DSL instead of querystring which is what the `params` parameter of `reindex` accepts.

Reply:
Another benefit of re-using `reindex` is e.g. that it has now been optimized with `.source(False)` to return only record (UU)IDs... Maybe we also need to move it under `_reindex` (similar to `_create`) so that it can be reused more easily.

Reply:
We can adapt reindex to take a DSL query instead or in addition to a query string. It hasn't really been used before so ok to refactor it to make it useful. It's fine to make things work first, and later move around

Reply:
done, made a `es_query` para optional